File: code/functions/pyprism/pyprism/data/tests/test_load_summary.py
# -*- coding: utf-8 -*-
"""
Created on Aug 18 2021

@author: Yoann Pradat

Tests for _load_summary.py module.
"""

from pyprism.data import load_summary_rna_fus, load_summary_rna_gex, load_summary_wes_mut

def test_load_summary_rna_gex_met500():
    df_sum = load_summary_rna_gex(study="met500", level="genes", metric="counts", test_mode=False)
    df_sum = load_summary_rna_gex(study="met500", level="genes", metric="counts", test_mode=True)
    df_sum = load_summary_rna_gex(study="met500", level="genes", metric="TPM", test_mode=False)
    df_sum = load_summary_rna_gex(study="met500", level="genes", metric="TPM", test_mode=True)


def test_load_summary_rna_fus_met500():
    df_sum = load_summary_rna_fus(study="met500", mode="arriba")
    df_sum = load_summary_rna_fus(study="met500", mode="ericscript")
    df_sum = load_summary_rna_fus(study="met500", mode="pizzly")
    df_sum = load_summary_rna_fus(study="met500", mode="starfusion")


def test_load_summary_rna_gex_prism():
    df_sum = load_summary_rna_gex(study="prism", level="genes", metric="counts", test_mode=False)
    df_sum = load_summary_rna_gex(study="prism", level="genes", metric="counts", test_mode=True)
    df_sum = load_summary_rna_gex(study="prism", level="genes", metric="TPM", test_mode=False)
    df_sum = load_summary_rna_gex(study="prism", level="genes", metric="TPM", test_mode=True)
    df_sum = load_summary_rna_gex(study="prism", level="transcripts", metric="counts", test_mode=False)
    df_sum = load_summary_rna_gex(study="prism", level="transcripts", metric="TPM", test_mode=False)


def test_load_summary_rna_fus_prism():
    df_sum = load_summary_rna_fus(study="prism", mode="arriba")
    df_sum = load_summary_rna_fus(study="prism", mode="ericscript")
    df_sum = load_summary_rna_fus(study="prism", mode="pizzly")
    df_sum = load_summary_rna_fus(study="prism", mode="starfusion")


def test_load_summary_rna_gex_tcga():
    df_sum = load_summary_rna_gex(study="tcga", level="genes", metric="counts", test_mode=False)
    df_sum = load_summary_rna_gex(study="tcga", level="genes", metric="counts", test_mode=True)
    df_sum = load_summary_rna_gex(study="tcga", level="genes", metric="TPM", test_mode=False)
    df_sum = load_summary_rna_gex(study="tcga", level="genes", metric="TPM", test_mode=True)


def test_load_summary_rna_fus_tcga():
    df_sum = load_summary_rna_fus(study="tcga", mode="aggregated")


def test_load_summary_rna_gex_tcga():
    df_sum = load_summary_rna_gex(study="tcga_6_samples", level="genes", metric="counts", other_mode="prism")
    df_sum = load_summary_rna_gex(study="tcga_6_samples", level="genes", metric="counts", other_mode="zheng")

File: code/functions/rprism/tests/testthat/test-load_summary.R

# RNA

test_that("load_summary_rna_fus() met500 all works", {
  skip_on_cran()

  df_summary <- load_summary_rna_fus(study="met500", mode="arriba")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_fus(study="met500", mode="pizzly")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_rna_gex() met500 all works", {
  skip_on_cran()

  df_summary <- load_summary_rna_gex(study="met500", level="genes", metric="counts")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_gex(study="met500", level="genes", metric="TPM")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_rna_gex() prism all works", {
  skip_on_cran()

  df_summary <- load_summary_rna_gex(study="prism", level="genes", metric="counts")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_gex(study="prism", level="genes", metric="TPM")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_gex(study="prism", level="transcripts", metric="counts")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_gex(study="prism", level="transcripts", metric="TPM")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_rna_fus() prism all works", {
  skip_on_cran()

  df_summary <- load_summary_rna_fus(study="prism", mode="ericscript")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_fus(study="prism", mode="starfusion")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_rna_gex() tcga all works", {
  skip_on_cran()

  df_summary <- load_summary_rna_gex(study="tcga", level="genes", metric="counts")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_rna_gex(study="tcga", level="genes", metric="TPM")
  expect_true(is(df_summary, "data.frame"))
})

# WES

test_that("load_summary_wes_mut() met500 all works", {
  skip_on_cran()

  df_summary <- load_summary_wes_mut(study="met500", mode="oncotator_unfiltered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="met500", mode="oncotator_filtered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="met500", mode="annovar_unfiltered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="met500", mode="annovar_pathogenic")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_wes_mut() prism all works", {
  skip_on_cran()

  df_summary <- load_summary_wes_mut(study="prism", mode="oncotator_unfiltered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="prism", mode="oncotator_filtered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="prism", mode="annovar_unfiltered")
  expect_true(is(df_summary, "data.frame"))

  df_summary <- load_summary_wes_mut(study="prism", mode="annovar_pathogenic")
  expect_true(is(df_summary, "data.frame"))
})


test_that("load_summary_wes_mut() tcga all works", {
  skip_on_cran()

  df_summary <- load_summary_wes_mut(study="tcga", mode="mc3_filtered")
  expect_true(is(df_summary, "data.frame"))
})
